frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,753 @@
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+  ValidationFailed = "validation - failed",
+}
+
+interface ValidationFailed {
+  type: FormActionType.ValidationFailed;
+  validationErrors: FormState["validationErrors"];
+}
+
+type FormAction =
+  | UpdateFieldAction
+  | SuccessAction
+  | FailAction
+  | SubmitAction
+  | ValidationFailed;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: Nullable<string>;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: null,
+        },
+        submitError: null,
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      const hasNoErrors = Object.values(state.validationErrors).every(
+        (prop) => prop == null || prop === ""
+      );
+
+      return {
+        ...state,
+        status: hasNoErrors ? FormStatus.Submitting : FormStatus.Idle,
+      };
+    }
+
+    case FormActionType.Success:
+      return {
+        // Display thanks page
+        ...state,
+        status: FormStatus.Submitted,
+      };
+
+    case FormActionType.Failed:
+      return {
+        // Resta in idle, setta errore e resetta i campi?
+        ...state,
+
+        submitError: "Error while sending your message",
+        status: FormStatus.Idle,
+      };
+
+    case FormActionType.ValidationFailed:
+      return {
+        ...state,
+        validationErrors: action.validationErrors,
+      };
+
+    default: {
+      assertNever(action);
+    }
+  }
+}
+
+export function ContactsForm() {
+  const [state, dispatch] = React.useReducer(reducer, blankForm);
+
+  function validateErrors(values: FormState["values"]) {
+    const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+
+    const validationErrors: FormState["validationErrors"] = {
+      name: null,
+      email: null,
+      message: null,
+    };
+
+    if (values.name.trim().length === 0) {
+      validationErrors.name = "Please insert your name";
+    } else {
+      validationErrors.name = null;
+    }
+
+    if (!regex.test(state.values.email)) {
+      validationErrors.email = "Please insert a valid email";
+    } else {
+      validationErrors.email = null;
+    }
+    if (state.values.message.trim().length == 0) {
+      validationErrors.message = "Please insert a message";
+    } else {
+      validationErrors.message = null;
+    }
+    return validationErrors;
+  }
```

Comment:
No need to declare this inside the component. Pure functions like this one should be declared outside the scope of the component that's using them